ls1intum/Artemis
# [Enhancement] Align buttons in some Views and add lock mechanism for Programming Assessment and some refactoring

### Checklist

### Motivation and Context



### Description

- Aligned buttons for programming assessment like other exercise types (exercise-assessment, exercise-scores, team-assessment)
- Implemented lock functionality for programming assessment, new submission and manual result are created during lock
- Refactored `code-editor-tutor-assessment.container` and `ProgrammingAssessmenResource` a bit

### Steps for Testing

Repeat this for normal programming exercise, team exercise and exam with programming exercise

1. Log in to Artemis
2. Navigate to Course Administration
3. Create programming exercise and participate and go to exercise assessment dashboard
4. Click on `Start New Assessment` button, student submission should be loaded and locked for you
4.1 Navigate back (back button or browser) DO NOT CLICK ON CANCEL, verify that is says now `Continue Assessment` and click
4.2 Click on cancel button, verify that it now says again `Start New Assessment`-> Submission is not locked anymore
4.3 Save/Submit assessment, after submitting button should change to `Open Assessment`
5. Go to scores page of the exercise and verify that for those submissions that have a manual assessment the button `View Assessment` is displayed
6. Manual assessment should work as always, check also if something is not working as it should

NOTE: Test also quickly assessment for modeling and file upload (I changed remove @Transactional from a call)

### Test Coverage



<img width="846" alt="image" src="https://user-images.githubusercontent.com/41108487/99458480-e60f6300-292c-11eb-82cb-311f8cbfff3b.png">


### Screenshots


Exercise Assessment (before, during and after assessing exercises)
![image](https://user-images.githubusercontent.com/41108487/99450094-9034be00-2920-11eb-947d-a4a2d3e9901f.png)

Exercise Scores view assessment button is only displayed when exercise has a manual result:
![image](https://user-images.githubusercontent.com/41108487/99449905-4c41b900-2920-11eb-8233-ebd3ccd68725.png)


Team Assessment:
Before assessing exercise teamProgramming2:
![image](https://user-images.githubusercontent.com/41108487/99448819-e5bc9b00-291f-11eb-9fcd-b68594f06296.png)

During assessment of exercise teamProgramming2 (after assessing it displays open assessment see teamProgramming2):
![image](https://user-images.githubusercontent.com/41108487/99449363-0be23b00-2920-11eb-8259-c893cd99fbb0.png)
